chalk-solve/src/infer/unify.rs

```diff
@@ -440,9 +452,23 @@ impl<'t, I: Interner> Unifier<'t, I> {
                 }
             }
 
+            // For now, the unevalutable unevaluated const is too generic.
+            (&ConstValue::Concrete(_), &ConstValue::Unevaluated(_))
+            | (&ConstValue::Unevaluated(_), &ConstValue::Concrete(_)) => Ok(self.goals.push(
```

Comment:
...and then to generate a subobligation here, kind of like associated type normalization. Is this planned for "future PRs" or is my idea of how to approach this outdated :)